### DISCUSSION TITLE
About how  to calculate channel matrix

### DISCUSSION THREAD

**Discussion by 571991380** [NVlabs/sionna#519] [Q&A]
Hello， i want to use sionna to calculate channel matrix.

h_freq = cir_to_ofdm_channel(frequencies, a, tau, normalize=True)
h_freq.shape：(1,1,8,4,1,14,128)
                        [batch size, num_rx, num_rx_ant, num_tx, num_tx_ant, num_time_steps, fft_size]

I think this result is too specific and has so many dimensions.
In this example, num_rx_ant=8, num_tx=4
I just want the final result dimension to be (8,4), so I wrote the following code：
h_freq=h_freq.numpy()
h = h_freq.mean(axis=(5, 6))
h = h.squeeze()

Is it okay to do this?

**Comment by jhoydis** [comment]
Hi @571991380,

No, that's probably not what you want to do. You are averaging the channel across OFDM symbols and subcarriers.
If you only need a single channel matrix, you can just pick one subcarrier and one OFDM symbol, e.g.,
``
h = h_freq[0,0,:,:,0,0,0]
``

However, in this case it is also useless to simulate more than one OFDM symbol and more than a single subcarrier.
